doc/man/man3/PlayerInputActions.3
.TH "PlayerInputActions" 3 "Version 1.1" "My Project" \" -*- nroff -*-
.ad l
.nh
.SH NAME
PlayerInputActions
.SH SYNOPSIS
.br
.PP
.PP
Inherits IInputActionCollection2, and IDisposable\&.
.SS "Classes"

.in +1c
.ti -1c
.RI "interface \fBIPlayerActions\fP"
.br
.ti -1c
.RI "struct \fBPlayerActions\fP"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBPlayerInputActions\fP ()"
.br
.ti -1c
.RI "void \fBDispose\fP ()"
.br
.ti -1c
.RI "bool \fBContains\fP (InputAction action)"
.br
.ti -1c
.RI "IEnumerator< InputAction > \fBGetEnumerator\fP ()"
.br
.ti -1c
.RI "void \fBEnable\fP ()"
.br
.ti -1c
.RI "void \fBDisable\fP ()"
.br
.ti -1c
.RI "InputAction \fBFindAction\fP (string actionNameOrId, bool throwIfNotFound=false)"
.br
.ti -1c
.RI "int \fBFindBinding\fP (InputBinding \fBbindingMask\fP, out InputAction action)"
.br
.in -1c
.SS "Properties"

.in +1c
.ti -1c
.RI "InputActionAsset \fBasset\fP\fR [get]\fP"
.br
.ti -1c
.RI "InputBinding? \fBbindingMask\fP\fR [get, set]\fP"
.br
.ti -1c
.RI "ReadOnlyArray< InputDevice >? \fBdevices\fP\fR [get, set]\fP"
.br
.ti -1c
.RI "ReadOnlyArray< InputControlScheme > \fBcontrolSchemes\fP\fR [get]\fP"
.br
.ti -1c
.RI "IEnumerable< InputBinding > \fBbindings\fP\fR [get]\fP"
.br
.ti -1c
.RI "\fBPlayerActions\fP \fBPlayer\fP\fR [get]\fP"
.br
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "IEnumerator IEnumerable\&. \fBGetEnumerator\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "readonly InputActionMap \fBm_Player\fP"
.br
.ti -1c
.RI "List< \fBIPlayerActions\fP > \fBm_PlayerActionsCallbackInterfaces\fP = new List<\fBIPlayerActions\fP>()"
.br
.ti -1c
.RI "readonly InputAction \fBm_Player_Move\fP"
.br
.ti -1c
.RI "readonly InputAction \fBm_Player_Jump\fP"
.br
.ti -1c
.RI "readonly InputAction \fBm_Player_Attack\fP"
.br
.ti -1c
.RI "readonly InputAction \fBm_Player_Run\fP"
.br
.in -1c
.SH "Constructor & Destructor Documentation"
.PP 
.SS "PlayerInputActions\&.PlayerInputActions ()"

.SH "Member Function Documentation"
.PP 
.SS "bool PlayerInputActions\&.Contains (InputAction action)"

.SS "void PlayerInputActions\&.Disable ()"

.SS "void PlayerInputActions\&.Dispose ()"

.SS "void PlayerInputActions\&.Enable ()"

.SS "InputAction PlayerInputActions\&.FindAction (string actionNameOrId, bool throwIfNotFound = \fRfalse\fP)"

.SS "int PlayerInputActions\&.FindBinding (InputBinding bindingMask, out InputAction action)"

.SS "IEnumerator< InputAction > PlayerInputActions\&.GetEnumerator ()"

.SS "IEnumerator IEnumerable\&. PlayerInputActions\&.GetEnumerator ()\fR [private]\fP"

.SH "Member Data Documentation"
.PP 
.SS "readonly InputActionMap PlayerInputActions\&.m_Player\fR [private]\fP"

.SS "readonly InputAction PlayerInputActions\&.m_Player_Attack\fR [private]\fP"

.SS "readonly InputAction PlayerInputActions\&.m_Player_Jump\fR [private]\fP"

.SS "readonly InputAction PlayerInputActions\&.m_Player_Move\fR [private]\fP"

.SS "readonly InputAction PlayerInputActions\&.m_Player_Run\fR [private]\fP"

.SS "List<\fBIPlayerActions\fP> PlayerInputActions\&.m_PlayerActionsCallbackInterfaces = new List<\fBIPlayerActions\fP>()\fR [private]\fP"

.SH "Property Documentation"
.PP 
.SS "InputActionAsset PlayerInputActions\&.asset\fR [get]\fP"

.SS "InputBinding? PlayerInputActions\&.bindingMask\fR [get]\fP, \fR [set]\fP"

.SS "IEnumerable<InputBinding> PlayerInputActions\&.bindings\fR [get]\fP"

.SS "ReadOnlyArray<InputControlScheme> PlayerInputActions\&.controlSchemes\fR [get]\fP"

.SS "ReadOnlyArray<InputDevice>? PlayerInputActions\&.devices\fR [get]\fP, \fR [set]\fP"

.SS "\fBPlayerActions\fP PlayerInputActions\&.Player\fR [get]\fP"


.SH "Author"
.PP 
Generated automatically by Doxygen for My Project from the source code\&.


Assets/Scripts/PlayerInputActions.cs
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was auto-generated by com.unity.inputsystem:InputActionCodeGenerator
//     version 1.7.0
//     from Assets/PlayerInputActions.inputactions
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine.InputSystem;
using UnityEngine.InputSystem.Utilities;

public partial class @PlayerInputActions: IInputActionCollection2, IDisposable
{
    public InputActionAsset asset { get; }
    public @PlayerInputActions()
    {
        asset = InputActionAsset.FromJson(@"{
    ""name"": ""PlayerInputActions"",
    ""maps"": [
        {
            ""name"": ""Player"",
            ""id"": ""4c401059-ba1b-4463-95c8-ad3a46a224c8"",
            ""actions"": [
                {
                    ""name"": ""Move"",
                    ""type"": ""PassThrough"",
                    ""id"": ""4305929f-a729-403a-8de4-ed4232bb33b9"",
                    ""expectedControlType"": ""Vector2"",
                    ""processors"": """",
                    ""interactions"": """",
                    ""initialStateCheck"": true
                },
                {
                    ""name"": ""Jump"",
                    ""type"": ""Button"",
                    ""id"": ""7974dede-2629-4330-a424-0049a69650ac"",
                    ""expectedControlType"": ""Button"",
                    ""processors"": """",
                    ""interactions"": """",
                    ""initialStateCheck"": false
                },
                {
                    ""name"": ""Attack"",
                    ""type"": ""Button"",
                    ""id"": ""1ed53120-44aa-4861-8c6d-a4a8210bcb1e"",
                    ""expectedControlType"": ""Button"",
                    ""processors"": """",
                    ""interactions"": """",
                    ""initialStateCheck"": false
                },
                {
                    ""name"": ""Run"",
                    ""type"": ""PassThrough"",
                    ""id"": ""c074b5b7-c8cc-40e8-b6e8-183aa53f0a1e"",
                    ""expectedControlType"": ""Button"",
                    ""processors"": """",
                    ""interactions"": ""Hold"",
                    ""initialStateCheck"": false
                }
            ],
            ""bindings"": [
                {
                    ""name"": ""WASD"",
                    ""id"": ""da6bd4e2-7dee-4d01-94c0-32ddf191211d"",
                    ""path"": ""2DVector(mode=2)"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Move"",
                    ""isComposite"": true,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": ""up"",
                    ""id"": ""3a253be7-01db-4e0c-b6bd-44e0227e3512"",
                    ""path"": ""<Keyboard>/w"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": ""down"",
                    ""id"": ""6788f571-465a-46f8-9c71-cb7af045a5a4"",
                    ""path"": ""<Keyboard>/s"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": ""left"",
                    ""id"": ""208ace16-f5f4-42c1-9170-0f8fb37504ac"",
                    ""path"": ""<Keyboard>/a"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": ""right"",
                    ""id"": ""1de04b34-9781-4551-b597-3dc7ff64f37e"",
                    ""path"": ""<Keyboard>/d"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": """",
                    ""id"": ""0853b2a4-b7a4-4899-a554-2d725652abea"",
                    ""path"": ""<Keyboard>/space"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Jump"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""8b305107-d775-4543-9241-150c3456f293"",
                    ""path"": ""<Mouse>/leftButton"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Attack"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""3eba3499-60eb-4f14-9d58-f1b65f1be9fc"",
                    ""path"": ""<Keyboard>/shift"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Run"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                }
            ]
        }
    ],
    ""controlSchemes"": []
}");
        // Player
        m_Player = asset.FindActionMap("Player", throwIfNotFound: true);
        m_Player_Move = m_Player.FindAction("Move", throwIfNotFound: true);
        m_Player_Jump = m_Player.FindAction("Jump", throwIfNotFound: true);
        m_Player_Attack = m_Player.FindAction("Attack", throwIfNotFound: true);
        m_Player_Run = m_Player.FindAction("Run", throwIfNotFound: true);
    }

    public void Dispose()
    {
        UnityEngine.Object.Destroy(asset);
    }

    public InputBinding? bindingMask
    {
        get => asset.bindingMask;
        set => asset.bindingMask = value;
    }

    public ReadOnlyArray<InputDevice>? devices
    {
        get => asset.devices;
        set => asset.devices = value;
    }

    public ReadOnlyArray<InputControlScheme> controlSchemes => asset.controlSchemes;

    public bool Contains(InputAction action)
    {
        return asset.Contains(action);
    }

    public IEnumerator<InputAction> GetEnumerator()
    {
        return asset.GetEnumerator();
    }

    IEnumerator IEnumerable.GetEnumerator()
    {
        return GetEnumerator();
    }

    public void Enable()
    {
        asset.Enable();
    }

    public void Disable()
    {
        asset.Disable();
    }

    public IEnumerable<InputBinding> bindings => asset.bindings;

    public InputAction FindAction(string actionNameOrId, bool throwIfNotFound = false)
    {
        return asset.FindAction(actionNameOrId, throwIfNotFound);
    }

    public int FindBinding(InputBinding bindingMask, out InputAction action)
    {
        return asset.FindBinding(bindingMask, out action);
    }

    // Player
    private readonly InputActionMap m_Player;
    private List<IPlayerActions> m_PlayerActionsCallbackInterfaces = new List<IPlayerActions>();
    private readonly InputAction m_Player_Move;
    private readonly InputAction m_Player_Jump;
    private readonly InputAction m_Player_Attack;
    private readonly InputAction m_Player_Run;
    public struct PlayerActions
    {
        private @PlayerInputActions m_Wrapper;
        public PlayerActions(@PlayerInputActions wrapper) { m_Wrapper = wrapper; }
        public InputAction @Move => m_Wrapper.m_Player_Move;
        public InputAction @Jump => m_Wrapper.m_Player_Jump;
        public InputAction @Attack => m_Wrapper.m_Player_Attack;
        public InputAction @Run => m_Wrapper.m_Player_Run;
        public InputActionMap Get() { return m_Wrapper.m_Player; }
        public void Enable() { Get().Enable(); }
        public void Disable() { Get().Disable(); }
        public bool enabled => Get().enabled;
        public static implicit operator InputActionMap(PlayerActions set) { return set.Get(); }
        public void AddCallbacks(IPlayerActions instance)
        {
            if (instance == null || m_Wrapper.m_PlayerActionsCallbackInterfaces.Contains(instance)) return;
            m_Wrapper.m_PlayerActionsCallbackInterfaces.Add(instance);
            @Move.started += instance.OnMove;
            @Move.performed += instance.OnMove;
            @Move.canceled += instance.OnMove;
            @Jump.started += instance.OnJump;
            @Jump.performed += instance.OnJump;
            @Jump.canceled += instance.OnJump;
            @Attack.started += instance.OnAttack;
            @Attack.performed += instance.OnAttack;
            @Attack.canceled += instance.OnAttack;
            @Run.started += instance.OnRun;
            @Run.performed += instance.OnRun;
            @Run.canceled += instance.OnRun;
        }

        private void UnregisterCallbacks(IPlayerActions instance)
        {
            @Move.started -= instance.OnMove;
            @Move.performed -= instance.OnMove;
            @Move.canceled -= instance.OnMove;
            @Jump.started -= instance.OnJump;
            @Jump.performed -= instance.OnJump;
            @Jump.canceled -= instance.OnJump;
            @Attack.started -= instance.OnAttack;
            @Attack.performed -= instance.OnAttack;
            @Attack.canceled -= instance.OnAttack;
            @Run.started -= instance.OnRun;
            @Run.performed -= instance.OnRun;
            @Run.canceled -= instance.OnRun;
        }

        public void RemoveCallbacks(IPlayerActions instance)
        {
            if (m_Wrapper.m_PlayerActionsCallbackInterfaces.Remove(instance))
                UnregisterCallbacks(instance);
        }

        public void SetCallbacks(IPlayerActions instance)
        {
            foreach (var item in m_Wrapper.m_PlayerActionsCallbackInterfaces)
                UnregisterCallbacks(item);
            m_Wrapper.m_PlayerActionsCallbackInterfaces.Clear();
            AddCallbacks(instance);
        }
    }
    public PlayerActions @Player => new PlayerActions(this);
    public interface IPlayerActions
    {
        void OnMove(InputAction.CallbackContext context);
        void OnJump(InputAction.CallbackContext context);
        void OnAttack(InputAction.CallbackContext context);
        void OnRun(InputAction.CallbackContext context);
    }
}
